pkg/networkmanager/interfaces.js

```diff
@@ -619,6 +619,26 @@ function NetworkManagerModel() {
                           };
         }
 
+        function JSON_parse_carefully(str) {
+            try {
+                return JSON.parse(str);
+            }
+            catch (e) {
+                return { };
```

Comment:
Yes, this happens when the sysadmin puts invalid JSON into /etc/sysconfig/network-scripts/ifcfg-foo, maybe via nmtui.

I have added a commit that draws some attention to that.